exporter/utils/node_details.py

```diff
@@ -0,0 +1,64 @@
+from exporter.osclient import get_ironic_client, session_adapter
+
+FREEPOOL_AGGREGATE_ID = 1
+
+
+def get_nodes(detail=False):
+    """Return list of ironic client node objects."""
+    ironic_client = get_ironic_client()
+    return ironic_client.node.list(detail=detail)
+
+
+def add_project_names(nodes):
+    """Add project names to list of ironic client node objects."""
+    nova_api = session_adapter('compute')
```

Comment:
Just curious, why not call Ironic in this way?

```python
ironic_api = session_adapter('baremetal')
```

Reply:
There are several ways to change the api version (need baremetal 1.29) with session adapter in theory, but none of them seemed to work. I found a max_version kwarg that you can pass that I probably should have tried.